safe-global/safe-wallet-web
# chore: storybook

This commit ads Storybook support: https://storybook.js.org

Files ending with .stories.tsx will appear inside of the storybook UI where we get preview for those UI components.

## What it solves
Currently it is very hard to figure out which component can be reused when we build a new feature. We have a lot of designs in figma, but it is hard to find out if we have implemented reusable components that could be used to speed up the development of a new feature. With storyboard we get a place where we can display those reusable components. In addition to this storyboard forces us to think in "composable components" starting from the smallest UI elements and composing up. One first things about the props a component should receive, builds the UI around it with the different possible stats and later just feeds this component with the necessary data.

## How to test it
Run 
1. yarn
2. yarn storybook -> this should open localhost preview of components for which we have a story. 

This is still very rough - for example the CopyButton component doesn't render fully correctly:
<img width="56" alt="grafik" src="https://github.com/safe-global/safe-wallet-web/assets/693770/bbf1bb6c-1ced-4191-809a-cb279c12bda7">

but this will be tackled in a separate PR.

## Screenshots
<img width="1277" alt="grafik" src="https://github.com/safe-global/safe-wallet-web/assets/693770/b6ff992c-aeb2-4f17-8110-5bc3d15286c9">

<img width="1259" alt="grafik" src="https://github.com/safe-global/safe-wallet-web/assets/693770/26ff9bed-bf46-48de-88a6-d07eaeac24e2">


## Checklist
